### ISSUE TITLE
Trying to retrieve the FEN for each move in a game

### ISSUE THREAD

**Issue by JeffML** [yo35/kokopu#43]
This returns FENs for moves in a game:
```js
    const fens = game.nodes().map(n=>n.position().fen())
```
The problem is this:  

**Game moves:**  1. d4 d5

**FENs returned:**
```
rnbqkbnr/pppppppp/8/8/3P4/8/PPP1PPPP/RNBQKBNR b KQkq - 0 1
rnbqkb1r/pppppppp/5n2/8/3P4/8/PPP1PPPP/RNBQKBNR w KQkq - 0 1
```

The second FEN should end in "0 2" (since it is the second ply).  I also notice that the _en passant_ indicator is always 0, which shouldn't be the case if, say, 1...Nf6 was black's response.

Is there any way to get an accurate FEN for a move in a game?

**Comment by yo35** [comment]
Indeed, there is currently no simple way to have the counters properly set in the FEN. There is a workaround for the move number counter:

```
games.nodes().map(n => n.position().fen({
  fullMoveNumber: n.moveColor() === 'w' ? n.fullMoveNumber() : n.fullMoveNumber() + 1,
}));
```

... but the halfmove clock (not the en *en passant* indicator, as you mention it) would still be 0.

To be fixed...

**Comment by JeffML** [comment]
Sorry, I meant to say 50-move counter, which I think refers to the same
field. En passant was a past issue elsewhere.

**Comment by yo35** [comment]
You'll achieve what you want with method [`Node.fen()`](https://kokopu.yo35.org/docs/current/classes/Node.html#fen), available in Kokopu version 4.8.0 (just released):

```
const fens = game.nodes().map(n => n.fen());
```

**Comment by JeffML** [comment]
Thanks for the tip!
